ocaml/idl/datamodel.ml

```diff
@@ -4823,7 +4823,9 @@ module VGPU_type = struct
         field ~qualifier:DynamicRO ~ty:(Set (Ref _gpu_group)) ~lifecycle:[Published, rel_vgpu_productisation, ""] "enabled_on_GPU_groups" "List of GPU groups in which at least one have this VGPU type enabled";
         field ~qualifier:StaticRO ~ty:implementation ~lifecycle:[Published, rel_dundee, ""] ~default_value:(Some (VEnum "passthrough")) "implementation" "The internal implementation of this VGPU type";
         field ~qualifier:StaticRO ~ty:String ~lifecycle:[Published, rel_dundee, ""] ~default_value:(Some (VString "")) "identifier" "Key used to identify VGPU types and avoid creating duplicates - this field is used internally and not intended for interpretation by API clients";
-        field ~qualifier: StaticRO ~ty:Bool ~lifecycle:[Published, rel_dundee, ""] ~default_value:(Some (VBool false)) "experimental" "Indicates whether VGPUs of this type should be considered experimental";
+        field ~qualifier:StaticRO ~ty:Bool ~lifecycle:[Published, rel_dundee, ""] ~default_value:(Some (VBool false)) "experimental" "Indicates whether VGPUs of this type should be considered experimental";
+        field ~qualifier:DynamicRO ~ty:(Set String) ~lifecycle:[Published, rel_plymouth, ""] ~default_value:(Some (VSet [])) "compatible_types_in_vm" "List of VGPU types which are compatible in one VM";
```

Comment:
I just noticed that the type here and below is `Set String`. I think we want `Set (Ref _vgpu_type)`.

Reply:
Yes, my first thought is also Ref vgpu_type, but actually we can't have the Ref before the vgpu_type created.
The compatible* were initialized during vgpu_type creation.
Like we would like to have A,B,C into type A's compatible list but actually B and C may not created now.

Reply:
You have to then create all of them first, and then fill in all of the compatible fields, which isn't too hard. It's not a problem yet anyway, because `A` will now only have `[A]` or `[]`.

Reply:
OK, I will make some changes in the coming commits. Thanks.